dcrsqlite/apisource.go

```diff
@@ -320,12 +317,83 @@ func (db *wiredDB) GetRawTransactionWithPrevOutAddresses(txid string) (*apitypes
 
 	msgTx := txhelpers.MsgTxFromHex(txhex)
 	if msgTx == nil {
-		return tx, prevOutAddresses, "", nil
+		return &apitypes.ExplorerTxData{
+			Tx:       tx,
+			VinAddrs: prevOutAddresses,
+		}
 	}
 
 	txType := txhelpers.DetermineTxTypeString(msgTx)
+	var txFee, txFeeRate float64
+	if txType == "Ticket" || (txType == "Regular" && len(tx.Vin[0].Coinbase) < 5) {
+		txFee = txhelpers.TxFee(msgTx)
+		txFeeRate = txhelpers.TxFeeRate(msgTx)
+	} else {
+		txFee = 0.0
+		txFeeRate = 0.0
+	}
+
+	vinfo := new(apitypes.VoteInfo)
+	if stake.DetermineTxType(msgTx) == stake.TxTypeSSGen {
+		validation, version, bits, choices, err := txhelpers.SSGenVoteChoices(msgTx, db.params)
+		if err != nil {
+			return &apitypes.ExplorerTxData{
```

Comment:
Since this is unexpected (vote with invalid vote bits, perhaps we should log it as Debug)